js/zb.js

```diff
@@ -2446,6 +2447,77 @@ module.exports = class zb extends Exchange {
         return result;
     }
 
+    async transfer (code, amount, fromAccount, toAccount, params = {}) {
+        await this.loadMarkets ();
+        const currency = this.currency (code);
+        let side = undefined;
+        if (fromAccount === 'spot' || toAccount === 'futures') {
+            side = 1;
+        } else {
+            side = 0;
+        }
+        const request = {
+            'currencyName': currency,
+            'amount': amount,
+            // 'clientId': params['clientId'], // "2sdfsdfsdf232342"
+            'side': side, // 1：Deposit (zb account -> futures account)，0：Withdrawal (futures account -> zb account)
+        };
+        const response = await this.contractV2PrivatePostFundTransferFund (this.extend (request, params));
+        //
+        //     {
+        //         "code": 10000,
+        //         "data": "2sdfsdfsdf232342",
+        //         "desc": "Success"
+        //     }
+        //
+        const timestamp = this.milliseconds ();
+        const transfer = {};
+        transfer['id'] = this.safeValue (response, 'data');
+        transfer['timestamp'] = timestamp;
+        transfer['datetime'] = this.iso8601 (timestamp);
+        transfer['currency'] = currency;
+        transfer['amount'] = amount;
+        transfer['fromAccount'] = fromAccount;
+        transfer['toAccount'] = toAccount;
+        transfer['status'] = this.safeValue (response, 'desc');
+        return this.parseTransfer (transfer, currency);
```

Comment:
I think that lines 2473-2483 could just be 

```
const data = this.safeValue (response, 'data');
return this.parseTransfer (data, currency);
```

@kroitor Or should it be `return this.parseTransfer (data, code);`?

Reply:
I would suggest to apply the same logic as we do in other similar cases, that is:

1. If the exchange's response is a single object – pass the entire response to the parsing method (to parseTransfer in this case).
```JavaScript
const response = await ...
return this.parseTransfer (response, currency);
```
2. If the exchange's response is an object inside an object:
```JavaScript
const response = await ...
const internalObject = this.safeValue (response, 'internalObject', {});
return this.parseTransfer (internalObject, currency);
```
↑ We add a default value of {} cause the subsequent parser will need an empty object or a non-empty object for further lookups for other fields within the parsed structure, so passing `undefined` to the parser would not work.
3. We do not fill the fields if they're missing in the exchange response. We can, however, fill them in for convenience of the user, but i would make that behavior optional. Most of the time, createOrder, withdraw, transfer and other methods that affect your balance – will guarantee to return a structure with an id in it, and, maybe, some other fields, but also could be nothing else but the id. The exchanges will then let you fetch the missing info by that id from a different endpoint like fetchOrder(s), fetchWithdrawal(s) or fetchTransfer(s).

Reply:
I made the transfer object instead of just returning the response because this is the response:
```
        //     {
        //         "code": 10000,
        //         "data": "2sdfsdfsdf232342",
        //         "desc": "Success"
        //     }
```
It doesn't have any useful information that is used by parseTransfer